In [74]:
import pandas as pd

In [3]:
data=pd.read_csv('train.csv')
data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 889 entries, 0 to 888
Data columns (total 12 columns):
PassengerId    889 non-null int64
Survived       889 non-null int64
Pclass         889 non-null int64
Name           889 non-null object
Sex            889 non-null object
Age            889 non-null float64
SibSp          889 non-null int64
Parch          889 non-null int64
Ticket         889 non-null object
Fare           889 non-null float64
Cabin          202 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.4+ KB


In [5]:
data.shape

(889, 12)

In [6]:
data.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         0
dtype: int64

In [7]:
from sklearn.preprocessing import LabelEncoder
encoder = LabelEncoder()

data['Sex'] = encoder.fit_transform(data['Sex'])
data['Embarked'] = encoder.fit_transform(data['Embarked'])

In [8]:
data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    1  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  38.0      1      0   
2                             Heikkinen, Miss. Laina    0  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  35.0      1      0   
4                           Allen, Mr. William Henry    1  35.0      0      0   

             Ticket     Fare Cabin  Embarked  
0         A/5 21171   7.2500   NaN         2  
1          PC 17599  71.2833   C85         0  
2  STON/O2. 3101282   7.9250   NaN         2  
3            113803  53.1000  C123         2  
4            373450   8.0500   NaN         2

In [9]:
data=data.drop(['Cabin','PassengerId','Name','Ticket'],axis=1)
data.head()

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0
2         1       3    0  26.0      0      0   7.9250         2
3         1       1    0  35.0      1      0  53.1000         2
4         0       3    1  35.0      0      0   8.0500         2

# Model 1

In [18]:
y= data.iloc[:,1:2]
y.head()

Pclass
0       3
1       1
2       3
3       1
4       3

In [15]:
x=data.drop(["Pclass"],axis=1)
x.head()

Survived  Sex   Age  SibSp  Parch     Fare  Embarked
0         0    1  22.0      1      0   7.2500         2
1         1    0  38.0      1      0  71.2833         0
2         1    0  26.0      0      0   7.9250         2
3         1    0  35.0      1      0  53.1000         2
4         0    1  35.0      0      0   8.0500         2

In [16]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(x,y, test_size = 0.3, random_state = 0)

In [19]:
from sklearn.naive_bayes import *
clf=BernoulliNB()

In [30]:
prediction = clf.fit(X_train,y_train).predict(X_test)

C:\Users\swapn\Anaconda3\lib\site-packages\sklearn\utils\validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [21]:
from sklearn.metrics import accuracy_score,confusion_matrix

In [28]:
print("The accuracy score is:",accuracy_score(y_test,prediction,normalize=True)*100)

The accuracy score is: 59.1760299625


In [26]:
print("The confusion matrix is:")
print(confusion_matrix(y_test,prediction))

The confusion matrix is:
[[ 30   8  32]
 [  9  10  30]
 [ 24   6 118]]


## Conclusion:

Out of total 267 records only 158 are successfully classified, giving a model accuracy of 59%.

# Model 2

In [43]:
y1 = data.iloc[:,2:3]
y1.head()

Sex
0    1
1    0
2    0
3    0
4    1

In [44]:
x1=data.drop(["Sex"],axis=1)
x1.head()

Survived  Pclass   Age  SibSp  Parch     Fare  Embarked
0         0       3  22.0      1      0   7.2500         2
1         1       1  38.0      1      0  71.2833         0
2         1       3  26.0      0      0   7.9250         2
3         1       1  35.0      1      0  53.1000         2
4         0       3  35.0      0      0   8.0500         2

In [45]:
X1_train,X1_test,y1_train,y1_test = train_test_split(x1,y1, test_size = 0.3, random_state = 0)

In [46]:
prediction1 = clf.fit(X1_train,y1_train).predict(X1_test)

C:\Users\swapn\Anaconda3\lib\site-packages\sklearn\utils\validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [48]:
print("The accuracy score is:",accuracy_score(y1_test,prediction1,normalize=True)*100)

The accuracy score is: 74.531835206


In [49]:
print("The confusion matrix is:")
print(confusion_matrix(y1_test,prediction1))

The confusion matrix is:
[[ 49  49]
 [ 19 150]]


# Conclusion

Out of total 267 records only 199 are successfully classified, giving a model accuracy of 74.53%.

# Model 3

In [50]:
y2 = data.iloc[:,4:5]
y2.head()

SibSp
0      1
1      1
2      0
3      1
4      0

In [51]:
x2=data.drop(["SibSp"],axis=1)
x2.head()

Survived  Pclass  Sex   Age  Parch     Fare  Embarked
0         0       3    1  22.0      0   7.2500         2
1         1       1    0  38.0      0  71.2833         0
2         1       3    0  26.0      0   7.9250         2
3         1       1    0  35.0      0  53.1000         2
4         0       3    1  35.0      0   8.0500         2

In [52]:
X2_train,X2_test,y2_train,y2_test = train_test_split(x2,y2, test_size = 0.3, random_state = 0)

In [53]:
prediction2 = clf.fit(X2_train,y2_train).predict(X2_test)
print("The accuracy score is:",accuracy_score(y2_test,prediction2,normalize=True)*100)

The accuracy score is: 68.9138576779


C:\Users\swapn\Anaconda3\lib\site-packages\sklearn\utils\validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [54]:
print("The confusion matrix is:")
print(confusion_matrix(y2_test,prediction2))

The confusion matrix is:
[[162  20   0   0   0   0   0]
 [ 43  22   0   0   0   0   0]
 [  6   2   0   0   0   0   0]
 [  5   2   0   0   0   0   0]
 [  2   0   0   0   0   0   0]
 [  1   0   0   0   0   0   0]
 [  2   0   0   0   0   0   0]]


# Conclusion

Out of total 267 records only 184 are successfully classified, giving a model accuracy of 69%

# Model 4

In [56]:
y3=data.iloc[:,5:6]
y3.head()

Parch
0      0
1      0
2      0
3      0
4      0

In [57]:
x3=data.drop(["Parch"],axis=1)
x3.head()

Survived  Pclass  Sex   Age  SibSp     Fare  Embarked
0         0       3    1  22.0      1   7.2500         2
1         1       1    0  38.0      1  71.2833         0
2         1       3    0  26.0      0   7.9250         2
3         1       1    0  35.0      1  53.1000         2
4         0       3    1  35.0      0   8.0500         2

In [58]:
X3_train,X3_test,y3_train,y3_test=train_test_split(x3,y3,test_size=0.3,random_state=0)

In [59]:
prediction3 = clf.fit(X3_train,y3_train).predict(X3_test)
print("The accuracy score is:",accuracy_score(y3_test,prediction3,normalize=True)*100)

The accuracy score is: 71.5355805243


C:\Users\swapn\Anaconda3\lib\site-packages\sklearn\utils\validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [60]:
print("The confusion matrix is:")
print(confusion_matrix(y3_test,prediction3))

The confusion matrix is:
[[182  16   0   0   0]
 [ 31   9   0   0   0]
 [ 24   3   0   0   0]
 [  1   0   0   0   0]
 [  1   0   0   0   0]]


# Conclusion

Out of total 267 records only 191 are successfully classified, giving a model accuracy of 72%.

# Model 5

In [61]:
y4=data.iloc[:,7:]
y4.head()

Embarked
0         2
1         0
2         2
3         2
4         2

In [62]:
x4=data.iloc[:,0:7]
x4.head()

Survived  Pclass  Sex   Age  SibSp  Parch     Fare
0         0       3    1  22.0      1      0   7.2500
1         1       1    0  38.0      1      0  71.2833
2         1       3    0  26.0      0      0   7.9250
3         1       1    0  35.0      1      0  53.1000
4         0       3    1  35.0      0      0   8.0500

In [70]:
X4_train,X4_test,y4_train,y4_test=train_test_split(x4,y4,test_size=0.3,random_state=0)

In [71]:
prediction4 = clf.fit(X4_train,y4_train).predict(X4_test)
print("The accuracy score is:",accuracy_score(y4_test,prediction4,normalize=True)*100)

The accuracy score is: 73.4082397004


C:\Users\swapn\Anaconda3\lib\site-packages\sklearn\utils\validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [72]:
print("The confusion matrix is:")
print(confusion_matrix(y4_test,prediction4))

The confusion matrix is:
[[  0   0  49]
 [  0   0  22]
 [  0   0 196]]


Conclusion

Out of total 267 records only 196 are successfully classified, giving a model accuracy of 73.40%